lgc/builder/DescBuilder.cpp

```diff
@@ -425,9 +425,8 @@ Value *BuilderImpl::getDescPtr(ResourceNodeType concreteType, ResourceNodeType a
     // - value for high 32 bits of the pointer; HighAddrPc to use PC
     if (node || topNode || concreteType != ResourceNodeType::DescriptorFmask) {
       unsigned shadowDescriptorTable = m_pipelineState->getOptions().shadowDescriptorTable;
-      bool shadow =
-          concreteType == ResourceNodeType::DescriptorFmask && shadowDescriptorTable != ShadowDescriptorTableDisable;
-      Value *highHalf = getInt32(shadow ? shadowDescriptorTable : HighAddrPc);
+      bool enableFmask = concreteType == ResourceNodeType::DescriptorFmask && m_pipelineState->getOptions().enableFmask;
+      Value *highHalf = getInt32(enableFmask ? shadowDescriptorTable : HighAddrPc);
```

Comment:
This doesn't make sense. We should only get here if fmask is enabled, and the check is about where to load the fmask from, so it should remain unchanged.